packages/rum-core/src/domain/contexts/pageStateContexts.ts

```diff
@@ -0,0 +1,141 @@
+import type { Duration, RelativeTime } from '@datadog/browser-core'
+import {
+  isExperimentalFeatureEnabled,
+  noop,
+  elapsed,
+  addDuration,
+  addEventListeners,
+  relativeNow,
+  DOM_EVENT,
+} from '@datadog/browser-core'
+
+export const PAGE_STATE_TIMELINE_MAX_LENGTH = 500
+
+export const enum PageState {
+  ACTIVE = 'active',
+  PASSIVE = 'passive',
+  HIDDEN = 'hidden',
+  FROZEN = 'frozen',
+  TERMINATED = 'terminated',
+}
+
+type PageStateTimeline = Array<{ state: PageState; startTime: RelativeTime }>
+
+type Counter = { count: number; duration: Duration }
+export type PageStateContext = {
+  active: Counter
+  passive: Counter
+  hidden: Counter
+  frozen: Counter
+  terminated: Counter
+}
+
+export interface PageStateContexts {
+  getPageStates: (startTime: RelativeTime, duration: Duration) => PageStateContext | undefined
+  stop: () => void
+}
+
+let pageStateTimeline: PageStateTimeline = []
+let state: PageState | undefined
+
+export function startPageStateContexts(): PageStateContexts {
+  if (!isExperimentalFeatureEnabled('resource_page_states')) {
+    return {
+      getPageStates: () => undefined,
+      stop: noop,
+    }
+  }
+
+  addPageState(getState())
+
+  const { stop } = addEventListeners(
+    window,
+    [
+      DOM_EVENT.PAGE_SHOW,
+      DOM_EVENT.FOCUS,
+      DOM_EVENT.BLUR,
+      DOM_EVENT.VISIBILITY_CHANGE,
+      DOM_EVENT.RESUME,
+      DOM_EVENT.FREEZE,
+      DOM_EVENT.PAGE_HIDE,
+    ],
+    (event) => {
+      if (!event.isTrusted) return
+
+      if (event.type === DOM_EVENT.FREEZE) addPageState(PageState.FROZEN)
+      else if (event.type === DOM_EVENT.PAGE_HIDE)
+        addPageState((event as PageTransitionEvent).persisted ? PageState.FROZEN : PageState.TERMINATED)
+      else addPageState(getState())
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I'm a bit surprised with the lack of parenthesis there. IMO, it could be easier to read with consistent line return.
 If it is not just me, should we enforce it?

Reply:
If everyone agree, I'll fix it and enforce it
cc @BenoitZugmeyer @yannickadam

Reply:
I would also expect curly brackets { } on the else if.